### PR TITLE
test(functional): create email fixtures for hooks

### PR DESCRIPTION
**Pull request by ashrivastava-qa**
## Because

- we want to isolate and assure proper set up and tear down for our email/accounts

## This pull request

- refactors common code in before, after and test methods to playwright test [fixtures](https://playwright.dev/docs/test-fixtures)

## Issue that this pull request solves

Closes: [FXA-8984](https://mozilla-hub.atlassian.net/browse/FXA-8984)

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

This is a new PR for the same requirements as https://github.com/mozilla/fxa/pull/16594. All the review comments have been incorporated in this new PR from the old PR. The only reason for discarding the old PR were the merge conflicts which were not getting resolved.

[FXA-8984]: https://mozilla-hub.atlassian.net/browse/FXA-8984?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/functional-tests/lib/fixtures/standard.ts
Comment (pdehaan) [inline packages/functional-tests/lib/fixtures/standard.ts:49]: Not sure if this is still needed since the `password` from L22 should be in scope.

File: packages/functional-tests/tests/oauth/oauthPermissions.spec.ts
Comment (pdehaan) [inline packages/functional-tests/tests/oauth/oauthPermissions.spec.ts:10]: This should use the imported `password` above from L5.
```suggestion
      emailOptions: [{ prefix: '', password }],
```

We might also be able to find a solution where `prefix` is just defaulted to `''` so we don't need to explicitly pass an empty string.

File: packages/functional-tests/tests/oauth/oauthPromptNone.spec.ts
Comment (pdehaan) [inline packages/functional-tests/tests/oauth/oauthPromptNone.spec.ts:10]:
```suggestion
      emailOptions: [{ prefix: '', password }],
```

File: packages/functional-tests/tests/oauth/oauthPromptNone.spec.ts
Comment (pdehaan) [inline packages/functional-tests/tests/oauth/oauthPromptNone.spec.ts:161]:
```suggestion
      emailOptions: [{ prefix: '', password }],
```

File: packages/functional-tests/tests/oauth/signUp.spec.ts
Comment (pdehaan) [inline packages/functional-tests/tests/oauth/signUp.spec.ts:12]:
```suggestion
        { prefix: '', password },
        { prefix: 'bounced{id}', password },
```

File: packages/functional-tests/tests/oauth/signinTokenCode.spec.ts
Comment (pdehaan) [inline packages/functional-tests/tests/oauth/signinTokenCode.spec.ts:11]:
```suggestion
      emailOptions: [{ prefix: 'sync{id}', password }],
```

File: packages/functional-tests/tests/oauth/syncSignIn.spec.ts
Comment (pdehaan) [inline packages/functional-tests/tests/oauth/syncSignIn.spec.ts:24]:
```suggestion
        { prefix: '', password },
        { prefix: 'sync{id}', password },
```

File: packages/functional-tests/tests/oauth/syncSignIn.spec.ts
Comment (pdehaan) [inline packages/functional-tests/tests/oauth/syncSignIn.spec.ts:77]:
```suggestion
      emailOptions: [{ prefix: 'sync{id}', password }],
```

File: packages/functional-tests/tests/postVerify/forcePasswordChange.spec.ts
Comment (pdehaan) [inline packages/functional-tests/tests/postVerify/forcePasswordChange.spec.ts:21]: We can reuse the `import`ed values from L8-9 above:
```suggestion
          password,
          newPassword,
```

File: packages/functional-tests/tests/postVerify/syncForcePasswordChange.spec.ts
Comment (pdehaan) [inline packages/functional-tests/tests/postVerify/syncForcePasswordChange.spec.ts:13]:
```suggestion
        { prefix: 'forcepwdchange{id}', password, newPassword },
```

File: packages/functional-tests/tests/postVerify/syncForcePasswordChange.spec.ts
Comment (pdehaan) [inline packages/functional-tests/tests/postVerify/syncForcePasswordChange.spec.ts:5]:
```suggestion
import { expect, test, password, newPassword } from '../../lib/fixtures/standard';
```

File: packages/functional-tests/tests/postVerify/syncForcePasswordChange.spec.ts
Comment (pdehaan) [inline packages/functional-tests/tests/postVerify/syncForcePasswordChange.spec.ts:7]: nit: We can delete this line and use the `import`ed value from the standard fixtures file.

File: packages/functional-tests/tests/signin/signinCached.spec.ts
Comment (pdehaan) [inline packages/functional-tests/tests/signin/signinCached.spec.ts:12]:
```suggestion
        { prefix: '', password },
        { prefix: 'sync{id}', password },
```

File: packages/functional-tests/tests/syncV3/fxDesktopHandshake.spec.ts
Comment (pdehaan) [inline packages/functional-tests/tests/syncV3/fxDesktopHandshake.spec.ts:16]:
```suggestion
        { prefix: '', password },
        { prefix: 'sync{id}', password },
```

File: packages/functional-tests/tests/syncV3/settings.spec.ts
Comment (pdehaan) [inline packages/functional-tests/tests/syncV3/settings.spec.ts:21]:
```suggestion
        { prefix: 'sync{id}', password, newPassword },
```

File: packages/functional-tests/tests/syncV3/settings.spec.ts
Comment (pdehaan) [inline packages/functional-tests/tests/syncV3/settings.spec.ts:101]:
```suggestion
      emailOptions: [{ prefix: 'sync{id}', password }],
```

File: packages/functional-tests/tests/syncV3/signinCached.spec.ts
Comment (pdehaan) [inline packages/functional-tests/tests/syncV3/signinCached.spec.ts:12]:
```suggestion
        { prefix: '', password },
        { prefix: 'sync{id}', password },
```

File: packages/functional-tests/lib/fixtures/standard.ts
Comment (pdehaan) [inline packages/functional-tests/lib/fixtures/standard.ts:161]: I'm not 100% sure about this&hellip; but I wonder if _this_ is where we can default the `prefix` to an empty string to make it a tiny bit DRYer.

```suggestion
        login.createEmail(emailOptions.prefix ?? '')
```

Now if the `emailOptions.prefix` is nullish it will just default to an empty string, as long as we make `prefix` optional above in the type definition.

File: packages/functional-tests/tests/oauth/oauthPermissions.spec.ts
Comment (pdehaan) [reply packages/functional-tests/tests/oauth/oauthPermissions.spec.ts:10]: Re: default prefix, I made a comment above at the place where I imagined one could possibly default to an empty string. If that actually works, this could just be simplified to this [since the prefix would be optional, like the `newPassword`]:

```js
      emailOptions: [{ password }],
```

File: packages/functional-tests/lib/fixtures/standard.ts
Comment (pdehaan) [inline packages/functional-tests/lib/fixtures/standard.ts:17]: Q: What if `prefix` is _optional_, and we default it to an empty string (see my ignorant comments below on L161).
```suggestion
  prefix?: string; // Prefix for the email address
```

File: packages/functional-tests/lib/fixtures/standard.ts
Comment (pdehaan) [reply packages/functional-tests/lib/fixtures/standard.ts:49]: Actually, I should probably do a big <kbd>git grep "passwordzxcv" packages/functional-tests</kbd> after we merge some of the suggestions/nits in this PR and we can confirm that we're only hardcoding that default password in roughly one place.

File: packages/functional-tests/tests/oauth/oauthPromptNone.spec.ts
Comment (ashrivastava-qa) [reply packages/functional-tests/tests/oauth/oauthPromptNone.spec.ts]: incorporated. Thanks!

File: packages/functional-tests/tests/oauth/signinTokenCode.spec.ts
Comment (ashrivastava-qa) [reply packages/functional-tests/tests/oauth/signinTokenCode.spec.ts]: incorporated. Thanks!

File: packages/functional-tests/tests/oauth/syncSignIn.spec.ts
Comment (ashrivastava-qa) [reply packages/functional-tests/tests/oauth/syncSignIn.spec.ts]: incorporated. Thanks!

File: packages/functional-tests/tests/oauth/syncSignIn.spec.ts
Comment (ashrivastava-qa) [reply packages/functional-tests/tests/oauth/syncSignIn.spec.ts]: incorporated. Thanks!

File: packages/functional-tests/tests/postVerify/forcePasswordChange.spec.ts
Comment (ashrivastava-qa) [reply packages/functional-tests/tests/postVerify/forcePasswordChange.spec.ts]: incorporated. Thanks!

File: packages/functional-tests/tests/postVerify/syncForcePasswordChange.spec.ts
Comment (ashrivastava-qa) [reply packages/functional-tests/tests/postVerify/syncForcePasswordChange.spec.ts]: incorporated. Thanks!

File: packages/functional-tests/tests/signin/signinCached.spec.ts
Comment (ashrivastava-qa) [reply packages/functional-tests/tests/signin/signinCached.spec.ts]: incorporated. Thanks!

File: packages/functional-tests/tests/syncV3/fxDesktopHandshake.spec.ts
Comment (ashrivastava-qa) [reply packages/functional-tests/tests/syncV3/fxDesktopHandshake.spec.ts]: incorporated. Thanks!

File: packages/functional-tests/tests/syncV3/settings.spec.ts
Comment (ashrivastava-qa) [reply packages/functional-tests/tests/syncV3/settings.spec.ts]: incorporated. Thanks!

File: packages/functional-tests/tests/syncV3/signinCached.spec.ts
Comment (ashrivastava-qa) [reply packages/functional-tests/tests/syncV3/signinCached.spec.ts]: incorporated. Thanks!

File: packages/functional-tests/lib/fixtures/standard.ts
Comment (ashrivastava-qa) [reply packages/functional-tests/lib/fixtures/standard.ts]: incorporated. Thanks!

File: packages/functional-tests/lib/fixtures/standard.ts
Comment (ashrivastava-qa) [reply packages/functional-tests/lib/fixtures/standard.ts]: incorporated. Thanks!

File: packages/functional-tests/tests/oauth/oauthPromptNone.spec.ts
Comment (ashrivastava-qa) [reply packages/functional-tests/tests/oauth/oauthPromptNone.spec.ts]: incorporated. Thanks!

File: packages/functional-tests/tests/oauth/signUp.spec.ts
Comment (ashrivastava-qa) [reply packages/functional-tests/tests/oauth/signUp.spec.ts]: incorporated. Thanks!

File: packages/functional-tests/tests/postVerify/syncForcePasswordChange.spec.ts
Comment (ashrivastava-qa) [reply packages/functional-tests/tests/postVerify/syncForcePasswordChange.spec.ts:5]: incorporated. Thanks!

File: packages/functional-tests/tests/postVerify/syncForcePasswordChange.spec.ts
Comment (ashrivastava-qa) [reply packages/functional-tests/tests/postVerify/syncForcePasswordChange.spec.ts:7]: incorporated. Thanks!

File: packages/functional-tests/lib/fixtures/standard.ts
Comment (ashrivastava-qa) [reply packages/functional-tests/lib/fixtures/standard.ts:49]: Made the changes.

**Comment by ashrivastava-qa** [comment]
@Trinaa I incorporated your feedback from the old PR for:
- Creating a reusable fixture that could be used for multiple email prefixes instead of the individual email fixture for each type to reduce duplication.
- Moving the 'password' const to the standard.ts file
- Checking for the account existence and then proceeding with the teardown steps.
- Catching specific exceptions in the teardown steps.

Please let me know if I missed anything. Thanks!

File: packages/functional-tests/tests/oauth/signinTokenCode.spec.ts
Comment (Trinaa) [inline packages/functional-tests/tests/oauth/signinTokenCode.spec.ts:39]: Could remove the assignment I think.
```suggestion
      await target.createAccount(syncEmail, password);
```

File: packages/functional-tests/tests/postVerify/forcePasswordChange.spec.ts
Comment (Trinaa) [inline packages/functional-tests/tests/postVerify/forcePasswordChange.spec.ts:28]: I think we can remove this assignment
```suggestion
      await target.auth.signUp(email, password, {
```

**Review by Trinaa — APPROVED** [review]
Pre-Approving with Comments
- Especially important is the comment in packages/functional-tests/tests/react-conversion/signup.spec.ts I think

Thank you for doing this work @ashrivastava-qa, you've consolidated a bunch of code which is very satisfying to see 🌟

File: packages/functional-tests/lib/fixtures/standard.ts
Comment (ashrivastava-qa) [reply packages/functional-tests/lib/fixtures/standard.ts:23]: Yes, it should be capitalized. Thanks for the catch!

File: packages/functional-tests/tests/syncV3/signIn.spec.ts
Comment (ashrivastava-qa) [reply packages/functional-tests/tests/syncV3/signIn.spec.ts:8]: Yes, you are right! Added the email fixture to all these files.

File: packages/functional-tests/tests/react-conversion/signup.spec.ts
Comment (ashrivastava-qa) [reply packages/functional-tests/tests/react-conversion/signup.spec.ts:4]: Yes agreed! Made the change!

File: packages/functional-tests/tests/syncV3/signinCached.spec.ts
Comment (ashrivastava-qa) [reply packages/functional-tests/tests/syncV3/signinCached.spec.ts]: I'd think so too ,but I will have to look at it thoroughly to test how the signUp is working.  I will check and let you know. Thanks!

**Pull request merged**
